client/app/components/ParameterValueInput.jsx

```diff
@@ -145,79 +143,49 @@ export class ParameterValueInput extends React.Component {
   }
 
   renderQueryBasedInput() {
-    const { value, onSelect, queryId, parameter } = this.props;
+    const { queryId, parameter } = this.props;
+    const { value } = this.state;
     return (
       <QueryBasedParameterInput
         className={this.props.className}
         parameter={parameter}
         value={value}
         queryId={queryId}
-        onSelect={onSelect}
+        onSelect={this.onSelect}
       />
     );
   }
 
   renderNumberInput() {
-    const { className, onSelect, applyButton } = this.props;
+    const { className } = this.props;
     const { value } = this.state;
-    const showApplyButton = applyButton && value !== this.props.value;
 
-    const onChange = (newValue) => {
-      this.setState({ value: newValue });
-      if (!applyButton) {
-        onSelect(newValue);
-      }
-    };
+    const normalize = val => !isNaN(val) && val || 0;
 
     return (
-      <div className="parameter-input-number" data-dirty={showApplyButton || null}>
-        <InputNumber
-          className={className}
-          value={!isNaN(value) && value || 0}
-          onChange={onChange}
-          onKeyUp={showApplyButton ? (e) => {
```

Comment:
@arikfr I removed the "enter" listener. Only "ctrl+enter" works. Lmk if you're ok with that.